sdk-core/src/main/java/care/data4life/sdk/migration/RecordCompatibilityService.kt

```diff
@@ -19,138 +19,101 @@ package care.data4life.sdk.migration
 import care.data4life.sdk.tag.Tags
-import io.reactivex.Observable
-import io.reactivex.Single
-import io.reactivex.functions.BiFunction
-import java.io.IOException
 
+// see: https://gesundheitscloud.atlassian.net/browse/SDK-572
 // see: https://gesundheitscloud.atlassian.net/browse/SDK-525
 @Migration("This class should only be used due to migration purpose.")
 class RecordCompatibilityService internal constructor(
-    private val apiService: NetworkingContract.Service,
-    private val tagCryptoService: TaggingContract.CryptoService,
     private val cryptoService: CryptoContract.Service,
-    private val tagHelper: TaggingContract.Helper = TagCryptoHelper
+    private val tagCryptoService: TaggingContract.CryptoService,
+    private val compatibilityEncoder: MigrationContract.CompatibilityEncoder = CompatibilityEncoder,
+    private val searchTagsBuilderBuilderFactory: NetworkingContract.SearchTagsBuilderFactory = SearchTagsBuilder
 ) : MigrationContract.CompatibilityService {
-    private fun encrypt(
-        plainTags: Tags,
-        plainAnnotations: Annotations
-    ): Pair<EncryptedTagsAndAnnotations, EncryptedTagsAndAnnotations> {
-        val tagEncryptionKey = cryptoService.fetchTagEncryptionKey()
-        return Pair(
-            tagCryptoService.encryptTagsAndAnnotations(
-                plainTags,
-                plainAnnotations,
-                tagEncryptionKey
-            ),
-            encryptTags(plainTags, tagEncryptionKey)
-                .also { encryptedTags ->
-                    encryptedTags.addAll(
-                        encryptAnnotations(plainAnnotations, tagEncryptionKey)
-                    )
-                }
-        )
-    }
-
-    @Throws(IOException::class)
-    private fun encryptTags(tags: Tags, tagEncryptionKey: GCKey): MutableList<String> {
-        return tags
-            .map { entry -> entry.key + TaggingContract.DELIMITER + tagHelper.normalize(entry.value) }
-            .let { normalizedTags ->
-                tagCryptoService.encryptList(
-                    normalizedTags,
-                    tagEncryptionKey
-                )
-            }
-    }
-
-    @Throws(IOException::class)
-    private fun encryptAnnotations(
-        annotations: Annotations,
-        tagEncryptionKey: GCKey
-    ): MutableList<String> {
+    private fun encryptTags(
+        tagEncryptionKey: GCKey,
+        tagGroupKey: String,
+        tagOrGroup: List<String>
+    ): List<String> {
         return tagCryptoService.encryptList(
-            annotations,
+            tagOrGroup.toList(),
             tagEncryptionKey,
-            ANNOTATION_KEY + TaggingContract.DELIMITER
+            tagGroupKey
         )
     }
 
-    private fun needsDoubleCall(
-        encodedAndEncryptedTags: List<String>,
-        encryptedTags: List<String>
-    ): Boolean = encodedAndEncryptedTags.sorted() != encryptedTags.sorted()
+    private fun addTags(
+        tagGroupKey: String,
```

Comment:
took me a while to understand this was the `key` part of the key value pair for the tags, i thought it was another encryption key.

Reply:
As mentioned above...any suggestions to improve it?

Reply:
labels would help, but also having key/value right after the other for instance.
Also i just notices, isnt tagOrGroup already a list? does it need a .toList() again?